src/expr.rs

```diff
@@ -1661,13 +1661,18 @@ fn rewrite_field(context: &RewriteContext,
                  offset: Indent)
                  -> Option<String> {
     let name = &field.ident.node.to_string();
-    let overhead = name.len() + 2;
+    let separator = if context.config.space_before_type_annotation {
+        " : "
+    } else {
+        ": "
+    };
```

Comment:
Can this be factored out into a function, I think you use the same idiom in the last PR?